File: Starter Problems 3/trip.py
op = open('tripin.txt', 'r')
n = int(op.readline())
data = []
for i in range(n):
	data.append(int(op.readline().strip()))


triples_pos = []

# Counting the triples and saving their positions in the list triples
for no in range(len(data)):
	if data[no] % 3 == 0:
		triples_pos.append(str(no + 1))

# Outputing answer
ot = open('tripout.txt', 'w')
if len(triples_pos) > 0:
	ot.write(str(len(triples_pos)) + '\n')
	for i in triples_pos:
		ot.write(str(i) + ' ')
else:
	ot.write('Nothing here!')

File: Starter Problems 3/trip.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    ifstream inFile;
    inFile.open("tripin.txt");
    inFile >> n;
    vector<int> positions;
    int temp;
    for (int i=0;i<n;i++) {
        inFile >> temp;
        if (temp % 3 == 0) {
            positions.push_back(i+1);
        }
    }
    sort(positions.begin(),positions.end());
    inFile.close();
    ofstream outFile;
    outFile.open("tripout.txt");
    if (positions.size() != 0) {
        outFile << positions.size() << "\n";
        for (vector<int>::iterator it = positions.begin();it != positions.end(); it++) {
            outFile << *it << " ";
        }
    }
    else {
        outFile << "Nothing here!";
    }
    outFile.close();
}
